dashboard/src/components/AppView/SecretItem.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from "react";
+import { Eye, EyeOff } from "react-feather";
+
+import { ISecret } from "../../shared/types";
+import "./SecretContent.css";
+
+interface ISecretItemProps {
+  secret: ISecret;
+}
+
+interface ISecretItemState {
+  display: { [s: string]: boolean };
+}
+
+class SecretItem extends React.Component<ISecretItemProps, ISecretItemState> {
+  public state: ISecretItemState = {
+    display: {},
+  };
+
+  public componentDidMount() {
+    const display = {};
+    Object.keys(this.props.secret.data).forEach(k => (display[k] = false));
+    this.setState(display);
+  }
+
+  public render() {
+    const { secret } = this.props;
+    let data: JSX.Element[] = [];
+    Object.keys(secret.data).forEach(k => {
+      data = data.concat(this.renderSecret(k));
```

Comment:
this does not seem to be just data anymore but rendered secret entries. `data` seems too generic.